### PR TITLE
block modular upgrades for tinkerbell

### PR DESCRIPTION
**Pull request by tatlat**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/api/v1alpha1/cluster_webhook.go [hunk 1]
```diff
@@ -470,6 +471,12 @@ func ValidateWorkerKubernetesVersionSkew(new, old *Cluster) field.ErrorList {
 	}
 	for _, nodeGroupNewSpec := range new.Spec.WorkerNodeGroupConfigurations {
 		newVersion := nodeGroupNewSpec.KubernetesVersion
+
+		if newVersion != nil && nodeGroupNewSpec.MachineGroupRef.Kind == TinkerbellMachineConfigKind {
```
Comment (vivek-koppuru) [inline pkg/api/v1alpha1/cluster_webhook.go:475]: Do we have this block for CLI already or is it not needed there?

Comment (tatlat) [reply pkg/api/v1alpha1/cluster_webhook.go:475]: although it's located in the webhook, this function is used in the cli as well

**Review by vivek-koppuru — APPROVED** [review]

File: pkg/api/v1alpha1/cluster_webhook.go (hunk 1 above)
Comment (vivek-koppuru) [reply pkg/api/v1alpha1/cluster_webhook.go:475]: Let's create an issue to move this out then as a followup.

**Pull request merged**